### PR TITLE
[DLRM]DLRM data processing

### PR DESCRIPTION
**Pull request by xuechendi**
1. need to work with dockerfile fix: https://github.com/intel/e2eAIOK/pull/15
2. verified with all 23 days data(w/ freq limit)
3. found there is a gap info on how to create train/valid/test from categorified data, @csdingbin
4. convert from parquet to binary -> spark will go OOM, let's directly use Bin's previous script

### CODE REVIEW

**Comment by xuechendi** [comment]
Can one of the admins verify this patch?

File: modelzoo/dlrm/data_processing/splitconversion.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+
+def process_convert(part_i, num_part, src_files, max_ind_range):
+    feat_dims = [0] * len(CAT_COLS)
+    print(f"Start to process {part_i + 1}/{num_part}")
+    t1 = timer()
+    pdf = pd.read_parquet(src_files)
+    pdf = pdf[sorted_column_name]
+    pdf[INT_COLS] = pdf[INT_COLS].astype(np.float32)
+    pdf[CAT_COLS] = pdf[CAT_COLS] % max_ind_range
```
Comment (xuechendi) [inline modelzoo/dlrm/data_processing/splitconversion.py:37]: @csdingbin , I put % max_ind_range here, in case you want to remove it

**Comment by xuechendi** [comment]
@csdingbin , I have added a split script and a convert script, so now following the README, one can start from downloading to complete create the binary and npz for training, processed data will be shared through email.

File: modelzoo/dlrm/data_processing/README.md [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+cd criteo/raw_data
+for i in `seq 0 23`;do gzip -d day_$i.gz & done
+ls
+day_0  day_10  day_12  day_14  day_16  day_18  day_2   day_21  day_23  day_4  day_6  day_8
+day_1  day_11  day_13  day_15  day_17  day_19  day_20  day_22  day_3   day_5  day_7  day_9
+```
+
+## Enter docker
+```
+docker run --shm-size=10g -it --privileged --network host --device=/dev/dri -v ${path_to_e2eaiok_dataset}:/home/vmagent/app/dataset -v `pwd`/:/home/vmagent/app/e2eaiok -w /home/vmagent/app/ xuechendi/oneapi-aikit:hydro.ai /bin/bash
```
Comment (csdingbin) [inline modelzoo/dlrm/data_processing/README.md:14]: Or need to use the old docker image? or just a typo?

Comment (xuechendi) [reply modelzoo/dlrm/data_processing/README.md]: just updated, should use docker file built e2eaiok-pytorch

**Comment by csdingbin** [comment]
Can we remove previous spark code under dlrm/preproc folder in your PR?

**Comment by Jian-Zhang** [comment]
is this the most recent once that delivers 0.8025 auc?

**Comment by xuechendi** [comment]
> is this the most recent once that delivers 0.8025 auc?

yes, just want to check with you on staff meeting, currently I used a faster way to do random merge of 23 days, while auc will go down to 0.8022. If we want to have 0.8025, I could switch to another method which will be a bit slower.

**Comment by xuechendi** [comment]
@Jian-Zhang @csdingbin , updated convert_to_binary.py and now score is back to 0.8025.  it will take about 5000 secs to download data from HDFS and doing zip merge to train_data.bin

**Pull request merged**